packages/sycamore-macro/src/component.rs

```diff
@@ -261,11 +261,31 @@ fn inline_props_impl(item: &mut ItemFn) -> Result<TokenStream> {
     let props_struct_ident = format_ident!("{}_Props", item.sig.ident);
 
     let inputs = item.sig.inputs.clone();
-    let props = inputs.into_iter().collect::<Vec<_>>();
+    let props = inputs
+        .into_iter()
+        .map(|arg| match arg {
+            FnArg::Typed(pat_type) => {
+                // Remove any mut or other pattern modifiers
+                let pat = match *pat_type.pat {
+                    Pat::Ident(mut ident_pat) => {
+                        ident_pat.mutability = None;
+                        Pat::Ident(ident_pat)
+                    }
+                    // Handle other patterns if necessary
+                    _ => panic!("Unsupported pattern type in inline_props"),
+                };
+                syn::PatType {
+                    pat: Box::new(pat),
+                    ..pat_type
+                }
+            }
+            FnArg::Receiver(_) => panic!("Components cannot have receivers"),
+        })
+        .collect::<Vec<_>>();
 
     let generics = &item.sig.generics;
     let generics_phantoms = generics.params.iter().enumerate().filter_map(|(i, param)| {
-        let phantom_ident = format_ident!("__phantom{i}");
+        let phantom_ident = format_ident!("__phantom{}", i);
```

Comment:
I'm not sure why you changed this

Reply:
> I'm not sure why you changed this

ok that was a mistake on my part ill fix it ASAP

Reply:
> I'm not sure why you changed this

i think that is the wrong way of formatting. `format_ident!("__phantom{i}")` treats` {i}` as a literal, so the identifier becomes `__phantom{i}` exactly, without replacing i with its value.

Reply:
I'm pretty sure the `{i}` works because it generates the right code. Also `format_ident!` uses `std::format!` under the hood and the new `format!()` syntax is supported since Rust 1.58.

Reply:
> I'm pretty sure the `{i}` works because it generates the right code. Also `format_ident!` uses `std::format!` under the hood and the new `format!()` syntax is supported since Rust 1.58.

im sorry i didnt know that 😓